zipline/pipeline/loaders/blaze/_core.pyx

```diff
@@ -436,12 +435,11 @@ cdef _array_for_column_impl(object dtype,
         )
     else:
         # this cast prevents a compiler crash
-        baseline_array = <object> out_array
+        baseline_array = out_array.view(dtype)
```

Comment:
Why this change?

Reply:
Oh, there were actually two other issues here:

1. we need to view the array as the correct type or datetime64[ns] and bool arrays will appear as ints
2. the unsafe optimization was actually shown to be unsafe in real data and the perf impact of removing it is fine